src/components/datepicker/localeUtils.ts

```diff
@@ -87,6 +87,19 @@ export const formatDate = (
 export const parseDate = (string: string, format = DateTime.DATE_SHORT as string, locale = defaultLocale) =>
   DateTime.fromFormat(string, format, { locale }).toJSDate();
 
+export const parseMultiFormatsDate = (value: string, formats: string[], locale: string) => {
+  let output = null;
+  formats.some((format) => {
```

Comment:
Because you you need access to the parsed values I don't think `some` (or any other higher order function) works very well here, I agree with Lowie, a simple for loop with early return would actually make most sense here.

Reply:
`some` has early return functionality, once `true` is returned and I am setting `output` variable inside that loop, (which I would also do in `for` loop), so what is advantage of using `for` instead of `some` here?

Reply:
I'm ok with using `some`, but I also agree with @lorgan3. 
You're not really using the return value of `some`,  but you're just using `return true` to stop the loop, and set a variable that is scoped outside of the loop.

Something like this makes more sense to me:

```ts
for (let i = 0; i < formats.length; i++) {
  const parsedDate = DateTime.fromFormat(value, formats[i], { locale });
  if (parsedDate.isValid) {
    return parsedDate.toJSDate();
  }
}

return null; // or just no return, if returning undefined is also fine
```